From ceb116f4ab3ad51f5fe7dd8cddd1d007c3f08121 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Fri, 28 Jun 2024 11:13:45 +0200
Subject: [PATCH] arm64: dts: st: add access controllers to ltdc node on
 stm32mp215

The display controller has several register areas/handles various
resources that represent several RIFSC entries.
Access for all of these resources must be checked when the RIFSC
bus is populated.

Change-Id: If9e70d70c28c8d182eb4bbb1c59c056575458205
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/390723
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp215.dtsi | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp215.dtsi b/arch/arm64/boot/dts/st/stm32mp215.dtsi
index 9f6aae3c2ef1..7aa6f2f25ca7 100644
--- a/arch/arm64/boot/dts/st/stm32mp215.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp215.dtsi
@@ -15,8 +15,22 @@ ltdc: display-controller@48010000 {
 		clocks = <&rcc CK_BUS_LTDC>, <&rcc CK_KER_LTDC>;
 		clock-names = "bus", "lcd";
 		resets = <&rcc LTDC_R>;
-		access-controllers = <&rifsc 80>;
 		power-domains = <&CLUSTER_PD>;
 		status = "disabled";
+		access-controllers = <&rifsc 80>;
+		access-controller-names = "cmn";
+
+		l0l1 {
+			access-controllers = <&rifsc 119>;
+			access-controller-names = "l0l1";
+		};
+		l2 {
+			access-controllers = <&rifsc 120>;
+			access-controller-names = "l2";
+		};
+		rot {
+			access-controllers = <&rifsc 121>;
+			access-controller-names = "rot";
+		};
 	};
 };
-- 
2.39.5

